isis/src/apollo/apps/apollofindrx/apollofindrx.cpp

```diff
@@ -169,15 +169,14 @@ namespace Isis {
         }
 
         if (cube->label()->hasObject("History")) {
-            PvlObject histObj = cube->label()->findObject("History");
             // Record apollofindrx history to the cube
+            QString histName = (QString)cube->label()->findObject("History")["Name"];
             // create a History Blob with value found in the History PvlObject's Name keyword
-            Isis::History histBlob( (QString)histObj["Name"] );
             // read cube's History PvlObject data into the History Blob
-            cube->read(histBlob);
+            Isis::History *hist = cube->readHistory(histName);
```

Comment:
I find it really weird that apollofindrx has different names for the history BLOB compared to everything else. Maybe we could take this out and use the normal logic?

Reply:
Maybe, I think leaving things as they are in the apps isn't the worst thing but I agree. Seems like there was some idea that history blobs could be stored in different locations on the label